### PR TITLE
Change control-plane-vnis property to leaf-list

### PR DESCRIPTION
**Pull request by mikewiebe**
### Change Scope

* Currently the `control-plane-vnis` property is of `type string` that represents the control-plane VNIs that are discovered by the control-plane behind a peer VTEP.  This string construct does not scale well if the number of VNI's is large and will make it cumbersome to parse a list of VNI's separated by some delimiter (like space or comma).  This proposal changes the property to a leaf-list of `type oc-evpn-types:vni-id`.

* Not backwards compatible change in v0 model

#### Cisco Implementation
  * https://www.cisco.com/c/en/us/products/collateral/switches/nexus-9000-series-switches/guide-c07-734107.html

```bash
n9kv-leaf1# show nve peers detail
Details of nve Peers:
----------------------------------------
Peer-Ip: 10.3.0.2
    NVE Interface       : nve1
    Peer State          : Up
    Peer Uptime         : 00:27:57
    Router-Mac          : 5e04.0000.1b08
    Peer First VNI      : 470000
    Time since Create   : 00:27:58
    Configured VNIs     : 4000,7000,470000
    Provision State     : peer-add-complete
    Learnt CP VNIs      : 4000,7000,470000
    vni assignment mode : SYMMETRIC
    Peer Location       : N/A
```

#### Juniper Implementation
 * https://www.juniper.net/documentation/us/en/software/junos/evpn-vxlan/topics/concept/evpn-vxlan-data-plane-encapsulation.html

#### Arista Show Command Output
  * Reference https://netmindblog.com/2019/01/25/arista-dci-evpnvxlan/
```bash
leaf2-dc1#show vxlan vni
VNI to VLAN Mapping for Vxlan1
VNI        VLAN        Source       Interface       802.1Q Tag 
---------- ----------- ------------ --------------- ---------- 
1000       1007*       evpn         Vxlan1          1007       
1100       100         static       Ethernet3       untagged   
1200       200         static       Ethernet4       untagged 
```

### Tree view
```
$ pyang -f tree -p release/models/*/* --tree-path=/network-instances/network-instance/connection-points
module: openconfig-network-instance
  +--rw network-instances
     +--rw network-instance* [name]
        +--rw connection-points
           +--rw connection-point* [connection-point-id]
              +--rw connection-point-id    -> ../config/connection-point-id
              +--rw config
              |  +--rw connection-point-id?   string
              +--ro state
              |  +--ro connection-point-id?   string
              +--rw endpoints
                 +--rw endpoint* [endpoint-id]
                    +--rw endpoint-id    -> ../config/endpoint-id
                    +--rw config
                            <...snip...>
                    +--ro state
                            <...snip...>
                    +--rw local
                            <...snip...>
                    +--rw remote
                            <...snip...>
                    +--rw vxlan
                       +--rw config
                            <...snip...>
                       +--ro state
                            <...snip...>
                       +--ro endpoint-peers
                       |  +--ro endpoint-peer* [peer-address]
                       |     +--ro peer-address    -> ../state/peer-address
                       |     +--ro state
                       |        +--ro peer-address?         oc-inet:ip-address
                       |        +--ro peer-state?           enumeration
                       |        +--ro uptime?               oc-types:timeticks64
                       |        +--ro control-plane-vnis*   oc-evpn-types:vni-id    <--- now a leaf list
                       |        +--ro router-mac?           oc-yang:mac-address
                       +--ro endpoint-vnis
                            <...snip...>
```

### CODE REVIEW

**Comment by mikewiebe** [comment]
@dplore I tried to open the `public-pr` job that is failing but it says I don't have permission to view it so I have no idea why it's failing

**Comment by wenovus** [comment]
Hey Mike, once you merge with master then it should fix the CI workflow.

File: release/models/network-instance/openconfig-evpn.yang [hunk 1]
```diff
@@ -40,9 +40,15 @@ module openconfig-evpn {
     domains, this is not currently supported and requires an extension
     of the model.";
 
-  oc-ext:openconfig-version "0.4.0";
+  oc-ext:openconfig-version "1.0.0";
```
Comment (dplore) [inline release/models/network-instance/openconfig-evpn.yang:43]: I suggest we only increment to 0.5.0, as the evpn model is probably not ready to be v1.0.0 yet?

**Comment by dplore** [comment]
To aid visualization I have included a tree with the change:

```
$ pyang -f tree -p release/models/*/* --tree-path=/network-instances/network-instance/connection-points
module: openconfig-network-instance
  +--rw network-instances
     +--rw network-instance* [name]
        +--rw connection-points
           +--rw connection-point* [connection-point-id]
              +--rw connection-point-id    -> ../config/connection-point-id
              +--rw config
              |  +--rw connection-point-id?   string
              +--ro state
              |  +--ro connection-point-id?   string
              +--rw endpoints
                 +--rw endpoint* [endpoint-id]
                    +--rw endpoint-id    -> ../config/endpoint-id
                    +--rw config
                            <...snip...>
                    +--ro state
                            <...snip...>
                    +--rw local
                            <...snip...>
                    +--rw remote
                            <...snip...>
                    +--rw vxlan
                       +--rw config
                            <...snip...>
                       +--ro state
                            <...snip...>
                       +--ro endpoint-peers
                       |  +--ro endpoint-peer* [peer-address]
                       |     +--ro peer-address    -> ../state/peer-address
                       |     +--ro state
                       |        +--ro peer-address?         oc-inet:ip-address
                       |        +--ro peer-state?           enumeration
                       |        +--ro uptime?               oc-types:timeticks64
                       |        +--ro control-plane-vnis*   oc-evpn-types:vni-id    <--- now a leaf list
                       |        +--ro router-mac?           oc-yang:mac-address
                       +--ro endpoint-vnis
                            <...snip...>
```

**Review by dplore — COMMENTED** [review]
For completeness, please provide reference (link to doc or example output) to two or more implementations which implement control-plane-vnis.  This is for ease of reviewers and makes it obvious for those less familiar with evpn that there should be a list of vni-id's versus a string.

**Comment by dplore** [comment]
@mikewiebe this is only waiting on you to provide some references to documentation or example output showing control plane vni's implemented as a list.

**Comment by mikewiebe** [comment]
> @mikewiebe this is only waiting on you to provide some references to documentation or example output showing control plane vni's implemented as a list.

Thanks @dplore.  I added the references above to the PR description.

**Comment by dplore** [comment]
LGTM.  Last call and will merge on May 16, 2023

**Review by dplore — APPROVED** [review]

**Pull request merged**